### DISCUSSION TITLE
supabase-js database query compare to another field

### DISCUSSION THREAD

**Discussion by gomflo86** [supabase/supabase#6005] [Questions] [answered]
It is possible to do this query with **supabase-js**?

```sql
SELECT * FROM products WHERE updated_at current_price < previous_price;
```

Already try this but did not work:

```javascript
 let { data: } = await supabase
    .from("products")
    .select("*")
    .lt("current_price", "previous_price")
```

**Comment by GaryAustin1** [comment]
I also answered this over in discord.
You will need to use an rpc call to a postgres function to do the sql and return setof records, or setof products.
You could also use a view.

**Reply by steve-chavez** [reply]
Suggesting a computed column as an alternative below(a bit easier to do/maintain than a rpc or view).

**Comment by steve-chavez** [comment, marked as answer]
Not directly but you can create a computed column:

```sql
create function price_up(products p) returns bool as $$
  select p.current_price < p.previous_price;
$$ language sql;
```

And then:

```js
 let { data: } = await supabase
    .from("products")
    .select("*")
    .is("price_up", true);
```

That looks clearer for an API as well(you can add a comment to this computed column so it shows in the OpenAPI output)

**Comment by sunilguptasg** [comment]
Would I have to do the same thing to compare two text columns? Meaning create a function for this?

If I execute this
```
create function uid_equal_to_frienduid(friends p) returns bool as $$
  select p.uid = p.frienduid;
$$ language sql;
```
I get an error "Failed to run sql query: type p does not exist"

**Reply by GaryAustin1** [reply]
I think it should be p friends.   That is var named p of type friends.
